bentoctl/cli/__init__.py

```diff
@@ -67,151 +54,123 @@ def bentoctl():
-        print_description(describe_deployment(file))
+        deployment_config = DeploymentConfig.from_file(file)
+        deployment_config.generate()
     except BentoctlException as e:
-        e.show()
+        console.print(f"[red]{e}[/]")
         sys.exit(1)
 
 
-@bentoctl.command(section=BentoctlSections.OPERATIONS)
-@click.option(
-    "--file",
-    "-f",
-    type=click.Path(exists=True),
-    help="The path to the deployment config file.",
-    required=True,
-)
-def describe(file):
+@bentoctl.command()
+@click.option("--deployment_config_file", "-f", help="path to deployment_config file")
+def generate(deployment_config_file):
     """
-    Shows the properties of the deployment given a deployment_config.
+    Generate the files for the deployment based on the template type provided.
     """
-    try:
-        print_description(describe_deployment(deployment_config_path=file))
-    except BentoctlException as e:
-        e.show()
-        sys.exit(1)
-
-
-@bentoctl.command(section=BentoctlSections.OPERATIONS)
-@click.option(
-    "--file",
-    "-f",
-    type=click.Path(exists=True),
-    help="The path to the deployment config file.",
-    required=True,
-)
-def update(file):
+    deployment_config = DeploymentConfig.from_file(deployment_config_file)
+    deployment_config.generate()
+
+
+@bentoctl.command()
+@click.argument("bento_tag")
+@click.option("--deployment_config_file", "-f", help="path to deployment_config file")
+@click.option("--push", is_flag=True, default=False)
+@click.option("--overwrite-deployable/--no-overwrite-deployable", default=True)
+@click.option("--registry-password")
+@click.option("--registry-username")
+def build(
+    bento_tag,
+    deployment_config_file,
+    push,
+    overwrite_deployable,
+    registry_username,
+    registry_password,
+):
     """
-    Update the deployment given a deployment_config.
+    Build the Docker image to deploy to the cloud. Pass the `--push` flag to push it
+    into the default registry.
     """
-    try:
-        update_deployment(deployment_config_path=file)
-        # describe the state of deployment
-        print_description(describe_deployment(file))
-    except BentoctlException as e:
-        e.show()
-        sys.exit(1)
+    deployment_config = DeploymentConfig.from_file(deployment_config_file)
+    deployment_config.set_bento(bento_tag)
+    (
+        dockerfile_path,
+        dockercontext_path,
+        build_args,
+    ) = deployment_config.create_deployable(
+        destination_dir=os.curdir,
+        overwrite_deployable=overwrite_deployable,
+    )
+    (
+        registry_url,
+        username,
+        password,
+        image_tag,
+    ) = deployment_config.authenticate_registry(registry_username, registry_password)
+
+    build_docker_image(
+        image_tag=image_tag,
+        context_path=dockercontext_path,
+        dockerfile=dockerfile_path,
+        additional_build_args=build_args,
+    )
```

Comment:
this should be determined by the push flag, if user decided not to push, we can also create a docker image file in the current directory (that could be a flag too)